tour_model_eval/Explore multiple datasets.ipynb

```diff
@@ -0,0 +1,947 @@
+    "    user_trip_list = [ecwe.Entry({\"data\": ecwct.Confirmedtrip(tr), \"_id\": tr[\"_id\"], \"metadata\": {\"key\": \"analysis/confirmed_trip\"}}) for tr in user_trip_df.to_dict(\"records\")]\n",
+    "    curr_sim = eamts.similarity(user_trip_list, FINAL_RADIUS, shouldFilter=FINAL_FILTER_OURSIM, cutoff=FINAL_CUTOFF_OURSIM)\n",
+    "    curr_sim.fit()\n",
+    "    trip_df.loc[user_trip_df.index, \"trip_cluster_oursim\"] = curr_sim.labels_.to_list()\n",
+    "    modeling_support_objects[user_id][\"similarity_model\"] = curr_sim\n",
+    "    return trip_df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d70afa9e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def h_score_no_na(labels_true, labels_pred):\n",
+    "    na_index = labels_true[pd.isna(labels_true)].index\n",
+    "    # Before we set the index to nan; we don't want to have a side effect here!\n",
+    "    new_labels_pred = labels_pred.copy()\n",
+    "    new_labels_pred.loc[na_index] = np.nan\n",
+    "    if (len(na_index) > 0):\n",
+    "        print(f\"Dropping nan indices {na_index} before calculating score\")\n",
+    "        # print(f\"{labels_true.dropna()}, {new_labels_pred.dropna()}\")\n",
+    "    return sm.homogeneity_score(labels_true.dropna(), new_labels_pred.dropna())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b8897a31",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def request_count(labels_pred):\n",
+    "    # once per real cluster\n",
+    "    # once per noisy point (since it is not in a cluster)\n",
+    "    # once per filtered trip (not really necessary for our current regime, but good to be prepared)\n",
+    "    return np.count_nonzero(labels_pred.unique() >= 0) \\\n",
+    "                    + np.count_nonzero(labels_pred == -1) \\\n",
+    "                    + np.count_nonzero(labels_pred == -2)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "governing-prophet",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def update_basic_stats(user_id, participant_df, trip_df):\n",
+    "    user_trip_df = trip_df[trip_df.user_id == user_id]\n",
+    "    basic_stats = {}\n",
+    "    basic_stats[\"n_labeled_trips\"] = len(user_trip_df)\n",
+    "    basic_stats[\"unique_label_combos\"] = list(user_trip_df.groupby([\"mode_confirm\", \"purpose_confirm\", \"replaced_mode\"]).groups)\n",
+    "    basic_stats[\"start_loc_in_cluster\"] = np.count_nonzero(user_trip_df.start_loc_cluster != -1)\n",
+    "    basic_stats[\"end_loc_in_cluster\"] = np.count_nonzero(user_trip_df.end_loc_cluster != 1)\n",
+    "    for algo in [\"dbscan\", \"oursim\"]:\n",
+    "        print(f\"Generating basic stats for {user_id} {len(user_trip_df)}, algo {algo} \")\n",
+    "        basic_stats[f\"n_trip_in_cluster_{algo}\"] = np.count_nonzero(user_trip_df[f\"trip_cluster_{algo}\"] != -1)\n",
```

Comment:
Here you use the same metric to count` n_trip_in_cluster` for DBSCAN and oursim. But oursim has -2 cluster. I think the result for oursim is not correct.

Reply:
`oursim` only has -2 cluster for the case in which we use cutoff. In this notebook, we only use `no_cutoff`
https://github.com/e-mission/e-mission-eval-private-data/pull/28/files/8f4df76544737505562e0dab8d96fcffc72cce3a#diff-5b27f01eda7481b2844df59e55fffa030ca22df0c58f9796b8d1bb7ad13b1089R180

So this is correct, even if it is not very meaningful, since it implies that `n_trip_in_cluster` = `n_trip` for the oursim case. That is also what you will see in the plots.